### PR TITLE
Add lists to when rules

### PR DESCRIPTION
**Pull request by ajmaddaford**
### What is the context of this PR?
Adds lists to when rules to enable routing logic based on the number of items in a list.

### How to review 
The `test_skip_condition_list.json` can be used to test the new functionality.

### CODE REVIEW

File: scripts/test_schemas.sh [hunk 1]
```diff
@@ -6,8 +6,8 @@ if [ "$1" == "--local" ] || [ "$2" == "--local" ]; then
 fi
 
 if [ "$run_docker" == true ]; then
-    docker pull onsdigital/eq-schema-validator:v3
-    validator="$(docker run -d -p 5001:5000 onsdigital/eq-schema-validator:v3)"
+    docker pull onsdigital/eq-schema-validator:add-lists-to-when-rules
```
Comment (JamesGardiner) [inline scripts/test_schemas.sh]: Reminder to remove before merge

**Review by JamesGardiner — APPROVED** [review]

File: app/questionnaire/rules.py [hunk 2]
```diff
@@ -257,5 +275,10 @@ def get_metadata_value(metadata, key):
     return metadata.get(key)
 
 
+def get_list_count(list_store, list_name):
+    list_values = list_store[list_name]
+    return len(list_values) if list_values else 0
```
Comment (MebinAbraham) [inline app/questionnaire/rules.py]: Since you always get an empty list back?
```suggestion
    return len(list_values)
```

**Review by MebinAbraham — CHANGES_REQUESTED** [review]
Minor comment otherwise looks good. 👍

File: app/questionnaire/rules.py (hunk 2 above)
Comment (ajmaddaford) [reply app/questionnaire/rules.py]: I've also removed `list_values` so this is now `return len(list_store[list_name])`.

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**